src/NLog/Targets/DatabaseParameterInfo.cs

```diff
@@ -98,6 +98,20 @@ public DatabaseParameterInfo(string parameterName, Layout parameterLayout)
         /// <docgen category='Parameter Options' order='10' />
         [DefaultValue(0)]
         public byte Scale { get; set; }
+
+        /// <summary>
+        /// Gets or sets the column type of the database.
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        [DefaultValue(null)]
+        public string DbType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the database column type property.
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        [DefaultValue(null)]
+        public string DbTypePropertyName { get; set; }
```

Comment:
Now that we have it generic it bothers me a bit to go back to having it hard coded for Oracle. What do you think about using reflection to find a public writable property of the given enum type then setting that property equal to the enum? That way we could eliminate the DbTypePropertyName configuration making configuration simpler. I'm thinking of doing it similar to how they set properties in this stack overflow answer: http://stackoverflow.com/a/824854/870152